packages/frontend/src/pages/Tile/components/TableBanner/ImportCsvButton.tsx

```diff
@@ -177,12 +190,17 @@ const ImportCsvModal = ({ onClose }: { onClose: () => void }) => {
             },
           },
           refetchQueries:
-            i === chunkedData.length - 1 ? [GET_ALL_ROWS] : undefined,
+            i === chunkedData.length - 1 && !onPreImport
+              ? [GET_ALL_ROWS]
+              : undefined,
           awaitRefetchQueries: true,
         })
         setRowsImported((i + 1) * CHUNK_SIZE)
       }
       setImportStatus('completed')
+      if (onPostImport) {
+        setTimeout(() => onPostImport(), 1000)
+      }
```

Comment:
can i check is the 1s timeout to ensure that the data has been imported before redirecting to the page? or is this for better UX for the user so that he doesn't immediately see the tiles page

Reply:
for ux purposes. to show user the completed dialog